lib/backend/buffer_test.go

```diff
@@ -134,11 +137,69 @@ func (s *BufferSuite) TestWatcherSimple(c *check.C) {
 	}
 }
 
+// TestWatcherCapacity checks various watcher capacity scenarios
+func (s *BufferSuite) TestWatcherCapacity(c *check.C) {
+	const gracePeriod = time.Second
+
+	ctx := context.TODO()
+	b := NewCircularBuffer(
+		BufferCapacity(1),
+		BacklogGracePeriod(gracePeriod),
+	)
+	defer b.Close()
+	b.SetInit()
+
+	w, err := b.NewWatcher(ctx, Watch{
+		QueueSize: 1,
+	})
+	c.Assert(err, check.IsNil)
+	defer w.Close()
+
+	select {
+	case e := <-w.Events():
+		c.Assert(e.Type, check.Equals, types.OpInit)
+	case <-time.After(100 * time.Millisecond):
```

Comment:
Is the timeout necessary? It doesn't look like there is room for flakiness here. I suggest we make this a single-case select and let `go test -timeout=...` deal with timing out the tests.

Same throughout.